src/middleware.ts

```diff
@@ -17,12 +18,13 @@ export default async function middleware(req: NextRequest) {
   );
   const isPublicRoute = publicRoutes.some((route) => path === route);
 
-  // 3. Get the session from the cookie
+  // 3. Get the session cookie - we only check for existence since the cookie is
+  // http-only and managed by the backend. If it exists and is invalid, the backend
+  // API calls will fail and trigger a logout.
   const sessionCookie = req.cookies.get("id");
-  const isValidSession = sessionCookie?.value;
+  const isValidSession = !!sessionCookie;
```

Comment:
Nice boolean type coercion before the boolean check here, I've not seen that done before and I learned something.

Reply:
sweet! I want to say this is something pretty idiomatic in javascript although I've seen it used in other languages as well, especially when things could be `null` or `nil` or `undefined`

Reply:
Yeah, it makes total sense in a more dynamic typing kind of situation. An operator-type coercion.